src/legacy/ui/public/directives/field_name/field_name_icon.tsx

```diff
@@ -17,62 +17,63 @@
  * under the License.
  */
 import React from 'react';
+import { palettes, EuiIcon } from '@elastic/eui';
+import { IconSize } from '@elastic/eui/src/components/icon/icon';
 
-interface Props {
-  type: string;
+interface IconMapEntry {
+  icon: string;
+  color: string;
+}
+interface FieldNameIconProps {
+  type:
+    | 'boolean'
+    | 'conflict'
+    | 'date'
+    | 'geo_point'
+    | 'geo_shape'
+    | 'ip'
+    | 'murmur3'
+    | 'number'
+    | '_source'
+    | 'string'
+    | string;
   label: string;
+  size?: IconSize;
+  useColor?: boolean;
 }
 
-export function FieldNameIcon({ type, label }: Props) {
-  switch (type) {
-    case 'boolean':
-      return <span aria-label={label} className="dscField__icon kuiIcon fa-adjust"></span>;
-
-    case 'conflict':
-      return <span aria-label={label} className="dscField__icon kuiIcon fa-warning"></span>;
-
-    case 'date':
-      return <span aria-label={label} className="dscField__icon kuiIcon fa-clock-o"></span>;
-
-    case 'geo_point':
-      return <span aria-label={label} className="dscField__icon kuiIcon fa-globe"></span>;
+const { colors } = palettes.euiPaletteColorBlind;
 
-    case 'geo_shape':
-      return <span aria-label={label} className="dscField__icon kuiIcon fa-globe"></span>;
+// defaultIcon => a unknown datatype
+const defaultIcon = { icon: 'questionInCircle', color: colors[0] };
 
-    case 'ip':
-      return <span aria-label={label} className="dscField__icon kuiIcon fa-laptop"></span>;
+const typeToEuiIconMap: Partial<Record<string, IconMapEntry>> = {
+  boolean: { icon: 'invert', color: colors[5] },
+  // icon for an index pattern mapping conflict in discover
+  conflict: { icon: 'alert', color: colors[8] },
+  date: { icon: 'calendar', color: colors[7] },
+  geo_point: { icon: 'globe', color: colors[2] },
+  geo_shape: { icon: 'globe', color: colors[2] },
+  ip: { icon: 'storage', color: colors[8] },
```

Comment:
Coincidentally, a new `ip` icon was recently added to EUI and will be available in the next releases. Let's remind one another to come back and swamp this one!